build_tools/cmake/build_all.sh

```diff
@@ -38,6 +38,12 @@ declare -a CMAKE_ARGS=(
   "-DCMAKE_INSTALL_PREFIX=$(realpath ${INSTALL_DIR})"
   "-DIREE_ENABLE_ASSERTIONS=${IREE_ENABLE_ASSERTIONS}"
 
+  # Enable HIP/ROCM and CUDA compiler and runtime.
+  "-DIREE_HAL_DRIVER_CUDA=ON"
+  "-DIREE_TARGET_BACKEND_CUDA=ON"
+  "-DIREE_HAL_DRIVER_HIP=ON"
+  "-DIREE_TARGET_BACKEND_ROCM=ON"
```

Comment:
These could follow how webgpu is set up above
```bash
# Enable WebGPU compiler builds and tests by default. All deps get fetched as
# needed, but some of the deps are too large to enable by default for all
# developers.
IREE_TARGET_BACKEND_WEBGPU_SPIRV="${IREE_TARGET_BACKEND_WEBGPU_SPIRV:-ON}"
```

For now I'd just match that style, but recent CI job refactoring work I've been doing may let us drop or dramatically simplify these `build_tools/cmake/build_*.sh` scripts soon 🤞 

* We'll have fewer monolithic builds that call for generic scripts getting reused across multiple CI jobs / platforms
* There will be less of a need for a single script to handle configure -> build `all` -> build `install` -> build `iree-test-deps`
* CMake presets could be used for settings like `lld` (see `new-linux-dev` in https://github.com/iree-org/iree/blob/main/build_tools/cmake/presets/new.json)